pulp_ansible/app/tasks/collections.py

```diff
@@ -503,6 +512,27 @@ async def run(self):
         """
         tasks = []
         loop = asyncio.get_event_loop()
+        root, api_version = await self._get_collection_api_root(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                pass
+
+            collection_info = self.collection_info or []
+            sources = {r.source for r in collection_info if r.source}
```

Comment:
Can this be unrolled to be more readable. I had to read it 3 times and while it's allowed I think we want code to be super readable.